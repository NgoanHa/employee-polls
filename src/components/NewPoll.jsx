import { useNavigate } from "react-router-dom";
import { connect } from "react-redux";
import { useState } from "react";
import { handleAddQuestion } from "../actions/questions";

const NewPoll = ({ dispatch }) => {

    const navigate = useNavigate();
    const [firstOption, setFirstOption] = useState("");
    const [secondOption, setSecondOption] = useState("");

    const handleFirstOptionValueChange = (e) => {
        const firstOptionValue = e.target.value;
        setFirstOption(firstOptionValue);
    };

    const handleSecondOptionValueChange = (e) => {
        const secondOptionValue = e.target.value;
        setSecondOption(secondOptionValue);
    };

    const handleSubmitAddQuestion = (e) => {
        e.preventDefault();
        dispatch(handleAddQuestion(firstOption, secondOption));
        navigate("/");
    };

    return (
        <div className="newpoll-container">
            <h1>New Poll</h1>
            <h3>Create Your Own Poll</h3>
            <form className="newpoll-form" onSubmit={handleSubmitAddQuestion}>
                <label>First Option</label>
                <div className="newpoll-form-input">
                    <input type="text" data-testid="firstOption" placeholder="First Option" value={firstOption} onChange={handleFirstOptionValueChange} required/>
                </div>
                <label>Second Option</label>
                <div className="newpoll-form-input">
                    <input type="text" data-testid="secondOption" placeholder="Second Option" value={secondOption} onChange={handleSecondOptionValueChange} required/>
                </div>
                <button data-testid="submit" type="submit">Submit</button>
            </form>
        </div>
    );
}
export default connect()(NewPoll);