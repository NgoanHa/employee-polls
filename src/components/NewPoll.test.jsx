import {fireEvent, render} from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter } from "react-router-dom";
import NewPoll from "./NewPoll";

describe("NewPoll", () => {
    it("NewPoll component is render", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPoll/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });

    it("value input is match", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPoll/>
                </BrowserRouter>
            </Provider>
        );

        const firstOption = component.getByTestId("firstOption");
        const secondOption = component.getByTestId("secondOption");

        fireEvent.change(firstOption, {target: {value: 'First Option'}});
        fireEvent.change(secondOption, {target: {value: 'Second Option'}});
        expect(firstOption.value).toBe("First Option");
        expect(secondOption.value).toBe("Second Option");
    });
});