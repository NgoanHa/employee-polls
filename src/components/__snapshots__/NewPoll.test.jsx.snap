// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll NewPoll component is render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="newpoll-container"
      >
        <h1>
          New Poll
        </h1>
        <h3>
          Create Your Own Poll
        </h3>
        <form
          class="newpoll-form"
        >
          <label>
            First Option
          </label>
          <div
            class="newpoll-form-input"
          >
            <input
              data-testid="firstOption"
              placeholder="First Option"
              required=""
              type="text"
              value=""
            />
          </div>
          <label>
            Second Option
          </label>
          <div
            class="newpoll-form-input"
          >
            <input
              data-testid="secondOption"
              placeholder="Second Option"
              required=""
              type="text"
              value=""
            />
          </div>
          <button
            data-testid="submit"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="newpoll-container"
    >
      <h1>
        New Poll
      </h1>
      <h3>
        Create Your Own Poll
      </h3>
      <form
        class="newpoll-form"
      >
        <label>
          First Option
        </label>
        <div
          class="newpoll-form-input"
        >
          <input
            data-testid="firstOption"
            placeholder="First Option"
            required=""
            type="text"
            value=""
          />
        </div>
        <label>
          Second Option
        </label>
        <div
          class="newpoll-form-input"
        >
          <input
            data-testid="secondOption"
            placeholder="Second Option"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          data-testid="submit"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
